pkg/operation/botanist/component/dependencywatchdog/bootstrap.go

```diff
@@ -413,27 +433,38 @@ func (b *bootstrapper) podLabels() map[string]string {
 	return nil
 }
 
-func (b *bootstrapper) containerCommand() []string {
+func (b *bootstrapper) containerArgs() []string {
 	switch b.values.Role {
-	case RoleEndpoint:
+	case RoleWeed:
 		return []string{
-			"/usr/local/bin/dependency-watchdog",
+			"weeder",
 			fmt.Sprintf("--config-file=%s/%s", volumeMountPath, configFileName),
-			"--deployed-namespace=" + b.namespace,
-			"--watch-duration=5m",
 		}
 
 	case RoleProbe:
 		return []string{
-			"/usr/local/bin/dependency-watchdog",
-			"probe",
+			"prober",
 			fmt.Sprintf("--config-file=%s/%s", volumeMountPath, configFileName),
-			"--deployed-namespace=" + b.namespace,
-			"--qps=20.0",
-			"--burst=100",
-			"--v=4",
+			"--kube-api-qps=20.0",
+			"--kube-api-burst=100",
 		}
 	}
 
 	return nil
 }
+
+func encodeWeederConfig(config *weederapi.Config) (string, error) {
```

Comment:
you could have just defined one method using golang generics instead

Reply:
yeah right, will do that.